samtranslator/model/apigateway.py

```diff
@@ -270,8 +270,22 @@ def _is_missing_identity_source(self, identity):
         query_strings = identity.get("QueryStrings")
         stage_variables = identity.get("StageVariables")
         context = identity.get("Context")
+        ttl = identity.get("ReauthorizeEvery")
 
-        if not headers and not query_strings and not stage_variables and not context:
+        required_properties_missing = not headers and not query_strings and not stage_variables and not context
+
+        try:
+            int(ttl)
+        # this will catch if ttl is None and not convertable to an int
+        except TypeError:
+            # previous behavior before trying to read ttl
+            if required_properties_missing:
+                return True
+
+            return False
+
+        # If we can resolve ttl, attempt to see if things are valid
+        if (ttl is None or int(ttl) > 0) and required_properties_missing:
```

Comment:
We already checked that `ttl` has an int value, checking `None` is unnecessary here. I think if we go with @mgrandis's suggestion, it will simplify some of the things here.

Reply:
See above comment about the above suggestion.